### PR TITLE
Change anomaly detector sample for java from track 1 to track2

### PR DESCRIPTION
**Pull request by hongli750210**
1. Change anomaly detector sample for java from track 1 to track2
2. Changed file:
    1)other-samples/java-sample/pom.xml
    2)other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java

@jongio for notification.

### CODE REVIEW

**Comment by jongio** [comment]
@conniey - Is this something you can review for us?  Thanks, Jon

File: other-samples/java-sample/pom.xml [hunk 1]
```diff
@@ -7,18 +7,29 @@
...
     <version>1.0</version>
-
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.0</version>
+                <configuration>
+                    <source>11</source>
```
Comment (conniey) [inline other-samples/java-sample/pom.xml:17]: By default, we target JDK8

File: other-samples/java-sample/pom.xml [hunk 2]
```diff
@@ -7,18 +7,29 @@
...
+        </plugins>
+    </build>
     <dependencies>
-        <dependency>
-            <groupId>org.apache.httpcomponents</groupId>
-            <artifactId>httpclient</artifactId>
-            <version>4.5.2</version>
-        </dependency>
         <dependency>
             <groupId>org.json</groupId>
```
Comment (conniey) [inline other-samples/java-sample/pom.xml:25]: You don't need to pull in this dependency. The track2 libraries have a dependency on Jackson which will easily deserialize the json file to an object.

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java [hunk 3]
```diff
@@ -1,76 +1,111 @@
...
+
+    private static final String CONTENT_TYPE = "Content-Type";
+
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final String SUBSCRIPTION_KEY = "<YOUR-SUBSCRIPTION-KEY>";
+
+    private static final String END_POINT = "<YOUR-ANOMALY-DETECTOR-END-POINT>";
+
+    private static final HttpPipelinePolicy AUTH_POLICY =
```
Comment (conniey) [inline other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java:44]: Is there a reason these all (AUTH_POLICY, ADD_HEADERS_POLICY, HTTP_PIPELINE) have to be static rather than using them inline in the main method?

This makes it really hard to follow.

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java [hunk 4]
```diff
@@ -1,76 +1,111 @@
...
+        AnomalyDetectorAsyncClient asyncClient = new AnomalyDetectorClientBuilder()
+                .endpoint(END_POINT).pipeline(HTTP_PIPELINE).buildAsyncClient();
+
+        JSONObject requestJson = new JSONObject(new JSONTokener((Main.class.getResourceAsStream("/request-data.json"))));
+        List<TimeSeriesPoint> timeSeriesPointList = new ArrayList<>();
+        JSONArray seriesArray = requestJson.getJSONArray("series");
+        for (int i = 0; i < seriesArray.length(); i++) {
+            JSONObject seriesObject = seriesArray.getJSONObject(i);
+            TimeSeriesPoint timeSeriesPoint = new TimeSeriesPoint()
+                    .setTimestamp(OffsetDateTime.ofInstant(Instant.parse(seriesObject.getString("timestamp")), ZoneOffset.UTC))
```
Comment (conniey) [inline other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java:65]: This can be simplified to:
Instant.parse(seriesObject.getString("timestamp")).atOffset(ZoneOffset.UTC);

Rather than using another static method.

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java [hunk 5]
```diff
@@ -1,76 +1,111 @@
...
+            System.out.println("**************************************************************");
+            System.out.println(String.format("Request url is : %s", response.getRequest().getUrl().toString()));
+            response.getRequest().getHeaders().forEach(httpHeader -> {
+                System.out.println("**************************************************************");
+                System.out.println(String.format("Request httpHeader name is : %s", httpHeader.getName()));
+                System.out.println(String.format("Request httpHeader Value is : %s", httpHeader.getValue()));
+            });
+            response.getRequest().getBody().subscribe(byteBuffer -> {
+                System.out.println("**************************************************************");
+                Charset charset = Charset.forName("UTF-8");
```
Comment (conniey) [inline other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java:85]: StandardCharsets.UTF_8 is more common.

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java [hunk 6]
```diff
@@ -1,76 +1,111 @@
...
+
+        DetectRequest request = new DetectRequest()
+                .setGranularity(TimeGranularity.fromString(requestJson.getString("granularity")))
+                .setSeries(timeSeriesPointList);
+
+        asyncClient.detectLastPointWithResponse(request).subscribe(response -> {
+            System.out.println("Request detail information...................................");
+            System.out.println("**************************************************************");
+            System.out.println(String.format("Request url is : %s", response.getRequest().getUrl().toString()));
+            response.getRequest().getHeaders().forEach(httpHeader -> {
```
Comment (conniey) [inline other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java:78]: There is no need to manually print these out. You can build the pipeline with `new HttpLogOptions().setLogLevel(HttpLogDetailLevel.HEADERS_AND_BODY)`?

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java [hunk 7]
```diff
@@ -1,76 +1,111 @@
...
-                    System.out.println(response.getStatusLine());
-                    System.out.println("Response content is :\n");
-                    System.out.println(EntityUtils.toString(respEntity, "utf-8"));
-                    System.out.println("----------");
+
+        DetectRequest request = new DetectRequest()
+                .setGranularity(TimeGranularity.fromString(requestJson.getString("granularity")))
+                .setSeries(timeSeriesPointList);
+
+        asyncClient.detectLastPointWithResponse(request).subscribe(response -> {
```
Comment (conniey) [inline other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java:74]: Is there a reason you want access to the HttpResponse object? We expect people to generally not care about that and instead, use `detectLastPoint`

**Review by conniey — CHANGES_REQUESTED** [review]

File: other-samples/java-sample/pom.xml (hunk 1 above)
Comment (hongli750210) [reply other-samples/java-sample/pom.xml:17]: JDK default version chenged to 8.

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java (hunk 3 above)
Comment (hongli750210) [reply other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java]: According to the comment, modified to using these all (AUTH_POLICY, ADD_HEADERS_POLICY, HTTP_PIPELINE) inline in the main method

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java (hunk 7 above)
Comment (hongli750210) [reply other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java]: According to the comment, invoke method changed to [ detectLastPoint ].

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java (hunk 5 above)
Comment (hongli750210) [reply other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java]: The process was deleted because the invoke method changed to [detectlastpoint].

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java (hunk 6 above)
Comment (hongli750210) [reply other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java]: The process was deleted because the invoke method changed to [detectlastpoint].

File: other-samples/java-sample/pom.xml (hunk 2 above)
Comment (hongli750210) [reply other-samples/java-sample/pom.xml]: According to the comment，fixed dependency package import settings in POM file

File: other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java (hunk 4 above)
Comment (hongli750210) [reply other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java]: According to the comment, it has been simplified to [ Instant.parse(series.timestamp()).atOffset(ZoneOffset.UTC) ].

**Comment by v-xuto** [comment]
@conniey Please re-review this PR.

**Comment by v-xuto** [comment]
@conniey We have updated by your comments, please re-review this PR.

**Comment by conniey** [comment]
@v-xuto It was easier for me to just create a PR into @hongli750210 's repository. We try not to add extra dependencies because it makes it harder to reason about when customers are trying to follow the sample.

Please review and merge.
https://github.com/hongli750210/AnomalyDetector/pull/1

**Comment by hongli750210** [comment]
@conniey Your fix has been merged into @hongli750210's repository.

**Review by conniey — APPROVED** [review]
